mutyper/cli.py

```diff
@@ -175,26 +174,29 @@ def variants(args):
         variant.INFO["mutation_type"] = mutation_type
         # ancestral allele
         AA = ancestor[variant.CHROM][variant.start].seq
+
         # polarize genotypes (and associated INFO) if alternative allele is
         # ancestral
         if variant.ALT[0] == AA:
             variant.INFO["AC"] = variant.INFO["AN"] - variant.INFO["AC"]
             variant.INFO["AF"] = variant.INFO["AC"] / variant.INFO["AN"]
-            # cyvcf2 docs say we need to reassign genotypes like this for the
-            # change to propagate (can't just update indexwise)
-            if variant.ploidy == 2:
-                # diploid
-                variant.genotypes = [
-                    [int(not gt[0]), int(not gt[1]), gt[2]] for gt in variant.genotypes
-                ]
-            elif variant.ploidy == 1:
-                # haploid
-                variant.genotypes = [
-                    [int(not gt[0]), gt[1]] for gt in variant.genotypes
-                ]
-            else:
+
+            if variant.ploidy not in valid_ploidy:
                 raise ValueError(f"invalid ploidy {variant.ploidy}")
 
+            # diploid or haploids
+            genotype_array = variant.genotype.array()
+            # checks that all genotype elements are from the set of {-1,0.1}
+            # each element in the last column is a 0,1 indicator for phasing status
+            unique_gts = set(np.unique(genotype_array[:, :-1]))
+            if not unique_gts <= set([-1, 0, 1]):
+                raise ValueError("invalid genotypes")
```

Comment:
should we move this above the if/else block, so the check occurs for all variants?

Reply:
you could also use an f string to print the invalid genotype, computed as set difference